internal/service/tesla.go

```diff
@@ -0,0 +1,212 @@
+				logger.Error().Err(cmErr).Msgf("GetVehicleByVIN: Failed to commit transaction for vehicle %s", vin)
+			}
+		}
+	}()
+
+	sd, err := dbmodels.SyntheticDevices(
+		dbmodels.SyntheticDeviceWhere.Vin.EQ(vin)).One(ctx, tx)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return nil, ErrVehicleNotFound
+		}
+		logger.Error().Err(err).Msgf("Failed to check if vehicle %s has been processed", vin)
+		return nil, err
+	}
+
+	return sd, nil
+}
+
+// GetByVehicleTokenID retrieves a vehicle by its VIN.
+func (ts *TeslaService) GetByVehicleTokenID(ctx context.Context, logger *zerolog.Logger, pdb *db.Store, tokenID int64) (*dbmodels.SyntheticDevice, error) {
+	tx, err := pdb.DBS().Writer.BeginTx(ctx, &sql.TxOptions{Isolation: sql.LevelReadCommitted})
+	if err != nil {
+		logger.Error().Err(err).Msgf("Failed to begin transaction for vehicle %d", tokenID)
+		return nil, err
+	}
+	defer func() {
+		if err != nil {
+			if rbErr := tx.Rollback(); rbErr != nil {
+				logger.Error().Err(rbErr).Msgf("GetVehicleByVIN: Failed to rollback transaction for vehicle %d", tokenID)
+			}
+		} else {
+			if cmErr := tx.Commit(); cmErr != nil {
+				logger.Error().Err(cmErr).Msgf("GetVehicleByVIN: Failed to commit transaction for vehicle %d", tokenID)
+			}
+		}
+	}()
+
+	sd, err := dbmodels.SyntheticDevices(
+		dbmodels.SyntheticDeviceWhere.VehicleTokenID.EQ(null.IntFrom(int(tokenID)))).One(ctx, tx)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return nil, ErrVehicleNotFound
+		}
+		logger.Error().Err(err).Msgf("Failed to check if vehicle %d has been processed", tokenID)
+		return nil, err
+	}
+
+	return sd, nil
+}
+
+// UpdateSubscriptionStatus updates the subscription status of the given SyntheticDevice.
+func (ts *TeslaService) UpdateSubscriptionStatus(ctx context.Context, synthDevice *dbmodels.SyntheticDevice, status string) error {
+	tx, err := ts.pdb.DBS().Writer.BeginTx(ctx, &sql.TxOptions{Isolation: sql.LevelReadCommitted})
+	if err != nil {
+		ts.logger.Error().Err(err).Msg("Failed to begin transaction for updating subscription status.")
+		return err
+	}
+	defer func() {
+		if err != nil {
+			if rbErr := tx.Rollback(); rbErr != nil {
+				ts.logger.Error().Err(rbErr).Msg("Failed to rollback transaction for updating subscription status.")
+			}
+		} else {
+			if cmErr := tx.Commit(); cmErr != nil {
+				ts.logger.Error().Err(cmErr).Msg("Failed to commit transaction for updating subscription status.")
+			}
+		}
+	}()
+
+	// Update subscription status
+	synthDevice.SubscriptionStatus = null.String{String: status, Valid: true}
+
+	// Save the changes to the database
+	_, err = synthDevice.Update(ctx, tx, boil.Infer())
+	if err != nil {
+		ts.logger.Error().Err(err).Msg("Failed to update synthetic device subscription status.")
+		return err
+	}
+
+	return nil
+}
+
+func DecisionTreeAction(fleetStatus *VehicleFleetStatus, vehicleTokenID int64) (*models.StatusDecisionResponse, error) {
+	var action models.StatusDecisionAction
```

Comment:
@elffjs @KevinJoiner - this is preliminary implementation of [the decision tree](https://app.excalidraw.com/l/2TO4srzVqui/9KBOyF5Ubbg) from our discussion. 
For messages "Streaming toggle not present. Start polling vehicle telemetry." and "Vehicle stream compatible. Subscribe to telemetry to enable streaming." let's prompt mobile up to the same endpoint - "/v1/tesla/telemetry/subscribe/%d" which internally will determine if the vehicle should start poll or start stream. 
The subscribe endpoint is under construction, but I want to see if we are aligned here.